### PR TITLE
Remove falling back to DefaultResolver

### PR DESCRIPTION
**Pull request by krisselden**
No one uses `MyApp.MyClass = Ember.Object.extend()` anymore, especially if they are using the ember-cli/resolver.  This removes extending from DefaultResolver since falling back to it ups the cost of misses.

### CODE REVIEW

File: addon/resolvers/classic/index.js [hunk 1]
```diff
@@ -108,17 +107,14 @@ function resolveOther(parsedName) {
     }
 
     return defaultExport;
-  } else {
-    return this._super(parsedName);
   }
 }
 
 // Ember.DefaultResolver docs:
```
Comment (knownasilya) [inline addon/resolvers/classic/index.js]: Probably remove these comments too?

**Comment by rwjblue** [comment]
The fallback behavior will need to issue a deprecation for a cycle before removal (in the next major bump of this lib).

**Comment by stefanpenner** [comment]
released as v5.0.1 🎉

**Comment by rwjblue** [comment]
After further discussion, I actually think this is just a straight up bugfix.  The scenario where the fallback was actually used is more likely to troll folks than do what you want.

Sorry for the run around @krisselden...

**Pull request merged**